client/ayon_usd/standalone/usd/pinning/__pinning_file_generation_funcs.py

```diff
@@ -0,0 +1,303 @@
+#     flattened_reference_list = [item for sublist in asset_paths for item in sublist]
+#     if not flattened_reference_list:
+#         return ref_list
+#
+#     for ref in flattened_reference_list:
+#
+#         if os.path.isabs(ref) or is_uri(ref):
+#             resolved_path = resolver.Resolve(ref)
+#         else:
+#             parent_file_dir_name = os.path.dirname(layer_path)
+#             abs_path = os.path.normpath(os.path.join(parent_file_dir_name, ref))
+#             resolved_path = resolver.Resolve(abs_path)
+#
+#         if isinstance(resolved_path, Ar.ResolvedPath):
+#             resolved_path = resolved_path.GetPathString()
+#
+#         ref_list.update(get_asset_dependencies(resolved_path, ref))
+#     return ref_list
+
+
+def write_pinning_file(
+    output_path: str,
+    pinning_data: Dict[str, str],
+    create_missing_dirs: bool = True,
+    overwrite_ok: bool = True,
+):
+    """writes out a pinning file to disk in the appropriate format
+
+    Args:
+        output_path (str): Path where the pinning file should be written to
+        pinning_data (str): The pinning dict that holds all the pinning data
+        create_missing_dirs (bool): Whether to ensure folders to output path
+            exist before writing the file.
+        overwrite_ok (bool): Whether to overwrite the pinning file if it already
+            exists.
+
+
+    Raises:
+        TypeError: raised if pinning_data is not a dict or output_path is not a JSON
+        FileExistsError: raised if output_path exists and overwrite_ok is False
+    """
+
+    # data validation
+    if not isinstance(pinning_data, dict):
+        raise TypeError("pinning data is not a dict")
+
+    if not output_path.endswith(".json"):
+        raise TypeError("output_path is not a json")
+
+    # file creatoin
+    if os.path.exists(output_path) and not overwrite_ok:
+        raise FileExistsError(f"Destination filepath already exists: {output_path}")
+
+    if create_missing_dirs:
+        os.makedirs(os.path.dirname(output_path), exist_ok=True)
+
+    with open(output_path, "w") as pinning_file:
+
+        write_data = {"ayon_resolver_pinning_data": pinning_data}
+        json.dump(write_data, pinning_file, indent=2)
+
+
+def generate_pinning_file(
+    entry_usd: str, root_info: Dict[str, str], pinning_file_path: str
+):
+
+    if not pinning_file_path.endswith(".json"):
+        raise RuntimeError(f"Pinning file path is not a json file {pinning_file_path}")
+
+    # Assume that the environment sets up the correct default AyonUsdResolver
+    resolver = Ar.GetResolver()
+    pinning_data = get_asset_dependencies(entry_usd, resolver)
+
+    root_less_pinning_file_data = remove_root_from_dependency_info(
+        pinning_data, root_info
+    )
+
+    root_less_pinning_file_data["ayon_pinning_data_entry_scene"] = _remove_sdf_args(
+        entry_usd
+    )
+    write_pinning_file(
+        os.path.join(pinning_file_path),
```

Comment:
This isn't joining anything?
```suggestion
        pinning_file_path,
```